great_expectations/datasource/data_connector/util.py

```diff
@@ -196,7 +196,7 @@ def _invert_regex_to_data_reference_template(
     regex_pattern: str,
     group_names: List[str],
 ) -> str:
-    """Create a string template based on a regex and corresponding list of group names.
+    r"""Create a string template based on a regex and corresponding list of group names.
```

Comment:
Is it necessary to make this a raw string? My familiarity with anything outside of `f-strings` is limited but I'm not sure this adds anything.

Reply:
Pyupgrade uses a raw string here because of the invalid \d escape sequences on line 204.  I believe that pytest will also raise warnings about these